pkg/pillar/cmd/upgradeconverter/upgradeconverter.go

```diff
@@ -114,7 +123,7 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject) in
 		log.Fatal(err)
 	}
 	log.Infof("Starting %s\n", ctx.agentName)
-	runHandlers(ctx)
+	runHandlers(ctx, preVaultconversionHandlers)
```

Comment:
I think we should make the phase explicit so that we can run "pre-vault" and "post-vault" both from the API and by invoking /opt/zededa/bin/upgradeconverter.
For the latter we can do that by having device-steps.sh call upgradeconverter pre-vault, and for the Run we should define a RunPreVaultHandlers function; the RunPre/RunPost can use a common runPhase function for the bulk of the work.

Reply:
Updated the changes accordingly, please check